apps/testing-javascript/src/templates/lesson-template.tsx

```diff
@@ -73,4 +77,158 @@ const LessonTemplate = () => {
   )
 }
 
+const checkIfGitHubUrlContainsBranch = (
+  github_branch_url: string | undefined,
+) => {
+  if (!github_branch_url) return false
+
+  const pattern = /\/tree\/.+$/
+
+  // Test if the URL contains a branch
+  return pattern.test(github_branch_url)
+}
+
+const CodeSandboxEmbed: React.FC<{url: string | undefined}> = ({url}) => {
+  if (!url) return null
+
+  const iframeRefCallback = function iframeRefCallback(iframe: any) {
+    if (iframe) {
+      iframe.contentWindow.location.replace(url)
+    }
+  }
+
+  return (
+    <div>
+      <iframe
+        ref={iframeRefCallback}
+        title="Code"
+        allowTransparency={true}
+        frameBorder="0"
+        scrolling="no"
+        width="100%"
+        height="550px"
+      />
+    </div>
+  )
+}
+
+const GitHubUrls: React.FC<{
+  branchUrl: string | undefined
+  diffUrl: string | undefined
+}> = ({branchUrl, diffUrl}) => {
+  if (!branchUrl) return null
+
+  // render one or both of the GitHub URLs
+  // return <div>GitHub URLs</div>
+  const message = `Get the code on GitHub`
+
+  const includesBranch = checkIfGitHubUrlContainsBranch(branchUrl)
+
+  return (
+    <div>
+      <div
+      // className={css`
+      //   color: ${colorValues['gray-darken-30']};
+      //   display: flex;
+      //   align-items: flex-start;
+      //   flex-direction: column;
+      //   font-family: ${fonts.regular};
+      // `}
```

Comment:
@DrShpongle I commented out these styles which were in the original codebase, but left them as reference. We should remove them before merging this branch.